test/upsert/mzcompose.py

```diff
@@ -240,3 +241,52 @@ def run_one_failpoint(c: Composition, failpoint: str, error_message: str) -> Non
 
     c.run("testdrive", "failpoint/05-reset.td")
     c.kill("clusterd1")
+
+
+def workflow_incident_49(c: Composition) -> None:
+    """Test creating sources in a remote clusterd process. lalala"""
+
+    c.down(destroy_volumes=True)
+
+    dependencies = [
+        "materialized",
+        "zookeeper",
+        "kafka",
+        "schema-registry",
+    ]
+
+    for (style, mz) in [
+        (
+            "with DISK",
+            Materialized(
+                options=[
+                    "--orchestrator-process-scratch-directory=/mzdata/source_data",
+                ],
+                additional_system_parameter_defaults={
+                    "upsert_source_disk_default": "true"
+                },
+                environment_extra=["MZ_PERSIST_COMPACTION_DISABLED=true"],
```

Comment:
@philip-stoev Is it ok to set this for all upsert tests?

Reply:
I think it would be prudent to test with and without compaction

Reply:
We could definitely just add 2 more combinations, but this will double the test runtime, which is why I'm hesitant. The change is fine as is I guess, and QA team can follow up on this. My suggestion for this would be to run only a part in normal tests, and move all combinations to nightly.

Reply:
These tests only take a few minutes to run! Most of the waiting is in the `rehydration` workflow, waiting for stats to be collected (we should probably make the stat collection interval configurable and more aggressive in CI)

Reply:
The upsert pipeline takes 6 min now, so if it starts taking 12, it will become one of the longer jobs in the CI so may eventually need to be trimmed down. So putting a separate step in Nightly would be best.

Reply:
I had the same feeling. I implemented it, will push once the test runs are through.